### PR TITLE
New config: More powerful :config- commands: remove

### PR DESCRIPTION
**Pull request by milogert**
- `:config-remove-list` command to remove items from a list.
- `:config-remove-dict` command to remove items from a dict.
- Test coverage.

Continues #2794

---
This change is [Reviewable](https://reviewable.io/reviews/qutebrowser/qutebrowser/4324)

### CODE REVIEW

File: qutebrowser/config/configcommands.py [hunk 1]
```diff
@@ -300,6 +300,59 @@ def config_add_dict(self, option, key, value, temp=False, replace=False):
...
 
+    @cmdutils.register(instance='config-commands')
+    @cmdutils.argument('option', completion=configmodel.list_option)
+    def config_remove_list(self, option, value, temp=False):
+        """Remove a value from a list.
+
+        Args:
+            option: The name of the option.
+            value: The value to remove from the list.
+            temp: Set value temporarily until qutebrowser is closed.
```
Comment (rcorre) [inline qutebrowser/config/configcommands.py]: **Remove**  value temporarily until qutebrowser is closed?

File: qutebrowser/config/configcommands.py [hunk 2]
```diff
@@ -300,6 +300,59 @@ def config_add_dict(self, option, key, value, temp=False, replace=False):
...
+
+    @cmdutils.register(instance='config-commands')
+    @cmdutils.argument('option', completion=configmodel.dict_option)
+    def config_remove_dict(self, option, key, temp=False):
+        """Remove a key from a dict.
+
+        Args:
+            option: The name of the option.
+            key: The key to remove from the dict.
+            temp: Set value temporarily until qutebrowser is closed.
```
Comment (rcorre) [inline qutebrowser/config/configcommands.py]: Remove value temporarily until qutebrowser is closed?

**Comment by The-Compiler** [comment]
Thanks! I agree with what @rcorre has said, fixed those up in 1c1223821c8fa165da4eccb754a334ad23f88055.

I also noticed some issues with the `configmodel` changes in the last PR, fixed in ee60159c165363878af352ef0163e32b8051b0c7 and 49850b61eddf37e298c454f0461d0f3583905199.

**Comment by rcorre** [comment]
I guess this is a little late because `config_add_dict` and `config_add_list` are already in, but would it make more sense to reverse the verb/noun?

- `config_add_dict` -> `config_dict_add`
- `config_remove_dict` -> `config_dict_remove`
- `config_add_list` -> `config_list_add`
- `config_remove_list` -> `config_list_remove`

`config_add_dict` sounds like you are adding a dict, rather than adding something to a dict.

**Comment by The-Compiler** [comment]
Done in 9d09aa1b405c14b67673d1dd067606b208293b7c.

**Pull request merged**